jbmc/src/java_bytecode/java_bytecode_language.cpp

```diff
@@ -899,6 +899,16 @@ bool java_bytecode_languaget::generate_support_functions(
   // parameter names
   convert_lazy_method(res.main_function.name, symbol_table_builder);
 
+  const symbolt &symbol =
+    symbol_table_builder.lookup_ref(res.main_function.name);
+  if(symbol.value.is_nil())
+  {
+    throw invalid_command_line_argument_exceptiont(
+      "the program has no entry point",
+      "function",
+      "Specify an entry point that is in the analysis context");
```

Comment:
Not sure most readers would comprehend "in the analysis context". How about "Specify an entry-point function defined on the classpath"? If they're using "--context-exclude" and so on they presumably understand the hole they're making in the realm of loaded classes?

Reply:
That would be misleading. The case that it is not on the classpath is already covered elsewhere. This only happens with context restrictions.

Reply:
Hmm. Perhaps name-check the offending `--context-exclude` argument?

Reply:
I don't understand

Reply:
For example, "The specified entry-point was not in the analysis context; check that it is included by your --context-include or --context-exclude options"

Reply:
Or even just "was not available"; I'm trying to avoid saying "the analysis context" and hoping they know what that means

Reply:
Ok, changed.